tests/authentication/test_authentication.py

```diff
@@ -218,7 +219,16 @@ def test_post_form_session_auth_passing_csrf(self):
         Ensure POSTing form over session authentication with CSRF token succeeds.
         Regression test for #6088
         """
-        from django.middleware.csrf import _get_new_csrf_token
+        # Remove this shim when dropping support for Django 2.2.
+        if django.VERSION < (3, 0):
+            from django.middleware.csrf import _get_new_csrf_token
+        else:
+            from django.middleware.csrf import (
+                _get_new_csrf_string, _mask_cipher_secret
+            )
+
+            def _get_new_csrf_token():
+                return _mask_cipher_secret(_get_new_csrf_string())
```

Comment:
I guess we *could* switch over to using the public `django.middleware.csrf.get_token(request)` function. (With a mocked out request instance?)

Having said that we were already relying on a bit of private implementation detail *before* this pull request, so this certainly isn't any worse - happy either way.

Reply:
🤷 I just went for the minimal change. 

I suspect if I saw `get_token(request)` there I'd need to go dig out the `csrf.py` source code to see what it does...

Reply:
Yeah I think minimal has to be better here, too. Yes it's private API, but then it *already was*.